test/e2e/node/kubelet.go

```diff
@@ -458,187 +458,144 @@ var _ = SIGDescribe("kubelet", func() {
 	// Tests for NodeLogQuery feature
 	ginkgo.Describe("kubectl get --raw \"/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=/<insert-log-file-name-here> [Feature:NodeLogQuery] [LinuxOnly]", func() {
 		var (
-			numNodes  int
-			nodeNames sets.String
+			numNodes       int
+			linuxNodeNames sets.String
 		)
 
 		ginkgo.BeforeEach(func(ctx context.Context) {
-			nodes, err := e2enode.GetBoundedReadySchedulableNodes(ctx, c, maxNodesToCheck)
+			nodes, err := e2enode.GetReadyNodesIncludingTainted(ctx, c)
+			if len(nodes.Items) == 0 {
+				framework.Fail("Expected at least one Linux node to be present")
+			}
 			numNodes = len(nodes.Items)
 			framework.ExpectNoError(err)
-			nodeNames = sets.NewString()
+			linuxNodeNames = sets.NewString()
 			for i := 0; i < numNodes; i++ {
-				nodeNames.Insert(nodes.Items[i].Name)
+				if nodes.Items[i].Status.NodeInfo.OperatingSystem == "linux" {
+					linuxNodeNames.Insert(nodes.Items[i].Name)
+				}
 			}
 		})
 
 		/*
-			Test if kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/?"
-			returns an error or not!
+			Test if kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query"
+			returns an error!
 		*/
 
-		ginkgo.It("should return the error of running without --query option", func() {
+		ginkgo.It("should return the error with an empty --query option", func() {
 			ginkgo.By("Starting the command")
 			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
 
-			for nodeName := range nodeNames {
-				queryCommand := "\"/api/v1/nodes/" + nodeName + "/proxy/logs/?"
+			for nodeName := range linuxNodeNames {
```

Comment:
Do we need to query all the linux nodes?

Reply:
I think testing one node is sufficient but I was just fixing the test. I am fine just picking one node and using that.

Reply:
We can do that in a separate PR too

Reply:
Given the linter-hints job was complaining about the Set, I restricted the test to one Linux node  in this PR itself.